### PR TITLE
Select menu options css tweak

### PR DESCRIPTION
**Pull request by shravnk**
- add to select component css
- define xh-select css vars

See #836

### CODE REVIEW

File: desktop/cmp/form/inputs/Select.scss [hunk 1]
```diff
@@ -86,14 +86,15 @@
     &__option {
       &--is-selected {
         color: var(--xh-text-color);
-        background: var(--xh-bg-alt);
+        background: var(--xh-select-selected);
```
Comment (TomTirapani) [inline desktop/cmp/form/inputs/Select.scss]: Do we *need* to declare new vars for this? Wouldn't switching the existing selected / focused styles be sufficient?
I'm wary of adding further vars if not necessary.

File: desktop/cmp/form/inputs/Select.scss [hunk 2]
```diff
@@ -86,14 +86,15 @@
...
       &--is-focused {
         color: var(--xh-text-color-highlight);
-        background: var(--xh-bg-highlight);
+        background: var(--xh-select-focused);
       }
     }
   }
 
 }
+//--xh-bg-highlight: var(--bg-highlight, #{mc-trans('blue', '200')});
```
Comment (TomTirapani) [inline desktop/cmp/form/inputs/Select.scss]: Remove this

File: styles/vars.scss [hunk 3]
```diff
@@ -192,6 +192,10 @@ body {
   --xh-scrollbar-bg: var(--scrollbar-bg, #{mc('blue-grey', '100')});
   --xh-scrollbar-thumb: var(--scrollbar-thumb, #{mc('blue-grey', '300')});
 
+  // Select menus
+  --xh-select-selected: var(--select-selected, #{mc('blue', '100')});
+  --xh-select-focused: var(--select-selected, #{mc('blue', '50')});
```
Comment (TomTirapani) [inline styles/vars.scss]: Here we should use `--select-focused` to allow the user to customise it.

**Comment by TomTirapani** [comment]
Some general thoughts in addition to the inline comments:
- If we use new vars, please add dark theme variants - currently hard to read in dark theme.
- Please apply the same fix to the mobile Select

**Comment by shravnk** [comment]
Thanks Tom, few thoughts:

- To avoid creating new variables, we could just flip the ones we had been using, i.e. 
selected -> `xh-bg-highlight` and focused -> `xh-bg-alt`. Also removes the need for dark mode styling . 

- I would suggest setting `!important` on the `is-selected` background (i.e. so that selected option does not change color when focused).

- Our code currently focuses the first option when the menu opens. That's normally fine, but can be confusing when there are two options and both are immediately shaded. One way to avoid this is to set a custom `renderValue` on `Select`:
`   @computed
    get renderValue() {
        return this.toInternal(this.externalValue)
    }`

Maybe overthinking that last point, but at least the first two CSS tweaks should offer an improvement.

**Comment by TomTirapani** [comment]
> To avoid creating new variables, we could just flip the ones we had been using

Yes, that's what I suggested. We can also then flip the order of `--is-selected` and `--is-focused`, so that the selected style overrides the focused style. Then we wouldn't need the `!important`.

Don't think we need to worry about your last point just yet - let's keep this a quick styling-only change.

**Comment by shravnk** [comment]
Right, sorry missed that top comment. Will amend the PR to only include CSS tweaks. Thank you.

**Pull request closed without merging**